### DISCUSSION TITLE
Is there a way to remove event listeners?

### DISCUSSION THREAD

**Discussion by TomzBench** [Microsoft/node-pty#612] [Q&A] [answered]
It seems like the Pty api hides the underlying event emitters and exposes `onData` and `onExit`  ...however, when I want to free up my terminal i would like to remove my `onData` and `onExit` event handlers so that they are not called again after I delete references to this terminal.
 

My motivation is that when managing a container of terminals, I am still getting an `onData` event even after calling `terminal.kill()` and deleteing the terminal.

So I need a way to prevent the `onData` event from firing after calling `kill` ...there are some issues around the `kill` portability i think.  But it would be nice to at least be able to free event listeners

**Comment by Tyriar** [comment, marked as answer]
All event listeners have a dispose method for disposing the listener:

```ts
const listener = pty.onData();
listener.dispose();
```

**Reply by TomzBench** [reply]
yep - that did it. thanks!
